course3/homework-unique_ptr/unique_ptr.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstddef>
+
+
+namespace karun
+{
+	template <class T>
+	class unique_ptr
+	{	
+		
+	public:
+		unique_ptr();
+		unique_ptr(std::nullptr_t);
```

Comment:
Shouldn't the explicit specifier prevent this implicit conversion from nullptr_t to T*, though? Or perhaps I misunderstand how explicit works.